pkg/api/interceptor.go

```diff
@@ -67,9 +67,23 @@ func (wa *WalletAuthorizer) Stream() grpc.StreamServerInterceptor {
 	}
 }
 
+func (wa *WalletAuthorizer) isProtocolVersion3(request *messagev1.PublishRequest) bool {
+	envelopes := request.Envelopes
+	if envelopes == nil || len(envelopes) == 0 {
+		return false
+	}
+	// If any of the envelopes are not for a v1 topic, then we treat the request as non-v1
+	for _, envelope := range envelopes {
+		if !strings.HasPrefix(envelope.ContentTopic, "/xmtp/3/") && !strings.HasPrefix(envelope.ContentTopic, "xmtp/3/") {
```

Comment:
The `xmtp/3` prefix variant should never really exist, or if messages come in that way then I think shouldn't be treated as a valid `/xmtp/3` protocol message. For example, if messages came in on `xmtp/0` the cleaner would wipe them after a day because it's not `/xmtp/0`